packages/test/src/test-integration-workflows.ts

```diff
@@ -204,3 +209,52 @@ test('Worker allows heartbeating activities after shutdown has been requested',
   await worker.run();
   t.is(cancelReason, 'CANCELLED');
 });
+
+export async function conditionTimeout0Simple(): Promise<boolean | undefined> {
+  let validationTimerFired = false;
+  // Don't lower this value, otherwise, in CI, there is risk that the timer fire
+  // while the WFT is being processed, which would result in an UnhandledCommand.
+  workflow
+    .sleep(10_000)
+    .then(() => (validationTimerFired = true))
+    .catch((e) => console.log(e));
+
+  return await workflow.condition(() => validationTimerFired, 0);
+}
+
+test('Internal patches does not cause non-determinism error on replay', async (t) => {
+  const { createWorker, executeWorkflow } = helpers(t);
+  const worker = await createWorker({
+    // Disable workflow caching, to force replay after the condition's sleep
+    maxCachedWorkflows: 0,
+  });
+  await worker.runUntil(async () => {
+    try {
+      await executeWorkflow(conditionTimeout0Simple);
+      t.pass();
+    } catch (e) {
+      t.fail((e as Error).message);
```

Comment:
Why catch instead of letting the test throw an exception with all of the context?